### DISCUSSION TITLE
defaultValue validation issue

### DISCUSSION THREAD

**Discussion by arun2git** [react-hook-form/react-hook-form#8663] [Q&A]
<Form onSubmit={handleSubmit(onSubmit)}>
              <Row>
                <Col>
                  <FormGroup>
                    <Label for="name">
                      Category<em>*</em>
                    </Label>
                    <input
                      className="form-control"
                      placeholder="Name"
                      defaultValue={provideData.category_name}
                      type="text"
                      {...register("category_name", {
                        required: true,
                        maxLength: 200,
                      })}
                    />
                    {errors.category_name && (
                      <p className="text-error">
                        Please check the Category Name
                      </p>
                    )}
                  </FormGroup>
                </Col>
              </Row>
              <Row>
                <Col>
                  <FormGroup>
                    <Label for="description">
                      Description<em>*</em>
                    </Label>
                    <textarea
                      className="form-control"
                      placeholder="Description"
                      defaultValue={provideData.category_description}
                      {...register("category_description", {
                        required: true,
                        maxLength: 400,
                      })}
                    ></textarea>
                    {errors.category_description && (
                      <p className="text-error">Please check the description</p>
                    )}
                  </FormGroup>
                </Col>
              </Row>
              <Row>
                <Col className="justify-content-end d-flex">
                  <Button className="add-btn" disabled={status}>
                    Edit Category
                  </Button>
                  {status && (
                    <FlashMessage duration={2000}>
                      <div className="message_api">
                        Category Edited Successfully
                      </div>
                    </FlashMessage>
                  )}
                </Col>
              </Row>
            </Form>

Actually I created edit form and I fetch default value with help of API, when I submit the form with defaultValue the validation error comes up, the default value is shown. How do I manage this in react. plz help

**Comment by Moshyfawn** [comment]
If your `defaultValues` come async from an API, make sure to call `reset` with the new `defaultValues` to update the form values, as the form is first initialized when you render the component with `useForm`hook, so without the async `defaultValues` that come after
```js
const { data } = useAsyncData()

const defaultValues = data || { field: "" }

const { reset } = useForm({ defaultValues })

useEffect(() => {
  if (defaultValues) {
    reset(defaultValues))
  }
}, [defaultValues, reset])
```

_As as side note, next time, please, spend a bit more time **at least** formatting your code example or even consider creating a reproduction codesandbox, as it's hard to read or understand your use-case other-wise, and we're not here to guess your issue / question or reproduce it for you ;)_
